src/Uplink/Auth/Provider.php

```diff
@@ -0,0 +1,92 @@
+<?php declare( strict_types=1 );
+
+namespace StellarWP\Uplink\Auth;
+
+use StellarWP\Uplink\Auth\Admin\Disconnect_Controller;
+use StellarWP\Uplink\Auth\Admin\Connect_Controller;
+use StellarWP\Uplink\Auth\Auth_Pipes\Multisite_Subfolder_Check;
+use StellarWP\Uplink\Auth\Auth_Pipes\Network_Token_Check;
+use StellarWP\Uplink\Auth\Auth_Pipes\User_Check;
+use StellarWP\Uplink\Auth\Token\Contracts\Token_Manager;
+use StellarWP\Uplink\Config;
+use StellarWP\Uplink\Contracts\Abstract_Provider;
+use StellarWP\Uplink\Pipeline\Pipeline;
+
+final class Provider extends Abstract_Provider {
+
+	/**
+	 * @inheritDoc
+	 */
+	public function register() {
+		if ( ! $this->container->has( Config::TOKEN_OPTION_NAME ) ) {
+			return;
+		}
+
+		$this->container->bind(
+			Token_Manager::class,
+			static function ( $c ) {
+				return new Token\Token_Manager( $c->get( Config::TOKEN_OPTION_NAME ) );
+			}
+		);
+
+		$this->register_nonce();
+		$this->register_authorizer();
+		$this->register_auth_disconnect();
+		$this->register_auth_connect();
+	}
+
+	private function register_nonce(): void {
+		/**
+		 * Filter how long the callback nonce is valid for.
+		 *
+		 * @note There is also an expiration time in the Uplink Origin plugin.
+		 *
+		 * Default: 35 minutes, to allow time for them to properly log in.
+		 *
+		 * @param int $expiration Nonce expiration time in seconds.
+		 */
+		$expiration = apply_filters( 'stellarwp/uplink/' . Config::get_hook_prefix() . '/auth/nonce_expiration', 2100 );
+		$expiration = absint( $expiration );
+
+		$this->container->singleton( Nonce::class, new Nonce( $expiration ) );
+	}
+
+	/**
+	 * Registers the Authorizer and the steps in order for the pipeline
+	 * processing.
+	 */
+	private function register_authorizer(): void {
+		$this->container->singleton(
+			Network_Token_Check::class,
+			static function ( $c ) {
+				return new Network_Token_Check( $c->get( Token_Manager::class ) );
+			}
+		);
+
+		$pipeline = ( new Pipeline( $this->container ) )->through( [
+			User_Check::class,
+			Multisite_Subfolder_Check::class,
+			Network_Token_Check::class,
+		] );
+
+		$this->container->singleton(
+			Authorizer::class,
+			static function () use ( $pipeline ) {
+				return new Authorizer( $pipeline );
+			}
+		);
+	}
+
+	private function register_auth_disconnect(): void {
+		add_action( 'admin_init', function(): void {
+			$this->container->get( Disconnect_Controller::class )->maybe_disconnect();
+		}, 9, 0 );
+	}
+
+	private function register_auth_connect(): void {
+		add_action( 'admin_init', function(): void {
+			$this->container->get( Connect_Controller::class )->maybe_store_token_data();
+		}, 9, 0 );
+	}
```

Comment:
Two things here.

1. Docblocks
2. To be a good WordPress citizen, these actions need to be unhookable, so we cannot use anonymous functions here. While we feel that this logic belongs in `admin_init`, an end user of our plugins might have a weird situation where that isn't viable, so providing a situation where they can unhook and re-hook to the action of their choosing is important.